runtime/lua/vim/lsp/client.lua

```diff
@@ -317,14 +323,14 @@ local function validate_config(config)
   validate('cmd_cwd', config.cmd_cwd, optional_validator(is_dir), 'directory')
   validate('cmd_env', config.cmd_env, 'table', true)
   validate('detached', config.detached, 'boolean', true)
-  validate('name', config.name, 'string', true)
-  validate('on_error', config.on_error, 'function', true)
+  validate('name', config.name, optional_validator('string'), 'string|false')
+  validate('on_error', config.on_error, optional_validator('function'), 'function|false')
   validate('on_exit', config.on_exit, { 'function', 'table' }, true)
   validate('on_init', config.on_init, { 'function', 'table' }, true)
   validate('on_attach', config.on_attach, { 'function', 'table' }, true)
   validate('settings', config.settings, 'table', true)
   validate('commands', config.commands, 'table', true)
-  validate('before_init', config.before_init, { 'function', 'table' }, true)
+  validate('before_init', config.before_init, optional_validator('function'), 'function|false')
```

Comment:
was 'table' in the old code a mistake?

Reply:
Yes, `before_init` can only be a function.